src/wallet/wallet.cpp

```diff
@@ -2261,11 +2261,17 @@ SpendableInputs CWallet::FindSpendableInputs(
 
                 // check to see if the coin conforms to the payment source
                 CTxDestination address;
+                bool hasDestination = ExtractDestination(output.scriptPubKey, address);
                 bool isSelectable =
-                    ExtractDestination(output.scriptPubKey, address) &&
-                    this->SelectorMatchesAddress(selector, address);
+                    hasDestination && this->SelectorMatchesAddress(selector, address);
                 if (isSelectable) {
```

Comment:
I would suggest eliminating `isSelectable` and just writing this as
```suggestion
                if (hasDestination && this->SelectorMatchesAddress(selector, address)) {
```